ethereum/eth/src/main/java/tech/pegasys/pantheon/ethereum/eth/sync/fastsync/FastSyncTargetManager.java

```diff
@@ -118,4 +118,9 @@ public boolean shouldSwitchSyncTarget(final SyncTarget currentTarget) {
   public boolean shouldContinueDownloading() {
     return !protocolContext.getBlockchain().getChainHeadHash().equals(pivotBlockHeader.getHash());
   }
+
+  @Override
+  public boolean isCaughtUpWithPeer(final EthPeer peer) {
+    return peer.chainState().getEstimatedHeight() < pivotBlockHeader.getNumber();
```

Comment:
This isn't the right thing to extract here.  It is the right condition for where it was (when fast syncing we can select any peer as our target as long as their chain height is above the pivot block).  We've caught up with that peer when our chain height >= pivotBlockHeader.getNumber().

Reply:
yeah. that's kinda obvious in retrospect.